src/cli/services/webapp-html-index-generator.js

```diff
@@ -26,40 +26,65 @@ _public.init = (config = {}) => {
   });
 };
 
-function buildAssetTags(paths = [], tagBuilderAction){
+function buildAssetTags(items = [], type){
   const tags = [];
-  paths.forEach(path => {
-    tags.push(tagBuilderAction(path));
+  items.forEach(item => {
+    tags.push(buildAssetTag(item, type));
   });
   return tags;
 }
 
-function buildLinkTag(path){
-  return `<link href="${buildExternalAssetPath(path)}?t=${Date.now()}" rel="stylesheet">`;
+function buildAssetTag(item, type){
+  const attrs = stringifyAssetTagAttrs(buildAssetTagAttrs(item, type));
+  return shouldUseScriptTag(item, type) ? `<script ${attrs}></script>` : `<link ${attrs}>`;
 }
 
-function buildScriptTag(path){
-  return `<script src="${buildExternalAssetPath(path)}?t=${Date.now()}"></script>`;
+function shouldUseScriptTag(item, type){
+  return type == 'script' && !isPreloadedAsset(item);
+}
+
+function buildAssetTagAttrs(item, type){
+  const attrs = typeof item == 'string' ? { [buildAssetPathAttrName(item, type)]: item } : item;
+  return shouldAppendStylesheetRelAttr(attrs, type) ? { ...attrs, rel: 'stylesheet' } : attrs;
+}
+
+function buildAssetPathAttrName(item, type){
+  return type == 'script' && !isPreloadedAsset(item) ? 'src' : 'href';
+}
+
+function isPreloadedAsset(item){
+  return ['preload', 'prefetch'].includes(item.rel);
+}
+
+function shouldAppendStylesheetRelAttr(attrs, type){
+  return !attrs.rel && type == 'stylesheet';
+}
+
+function stringifyAssetTagAttrs(attrs){
+  return Object.entries(attrs).map(([key, value]) => {
+    return value ? `${key}="${formatExternalAssetAttrValue(key, value)}"` : key;
```

Comment:
Should this encode the value (potentially also the key, but mostly the value)?

Reply:
It don't think it's necessary for now. Do you have some *use case* in mind which it would be important for?

Reply:
Since it'd be most likely a developer controlling all these values, nothing in particular comes to mind that would be important to address.

Reply:
> Since it'd be most likely a developer controlling all these values

Exactly. It seems safe enough 👍